### PR TITLE
Remove version.qualified from MainResponse

### PR DESCRIPTION
**Pull request by alpar-t**
Concluding the discussions after merging #35334. 
`version.qualified` is removed from the main response and 
the fully qualified version will be returned as `version.number`.

### CODE REVIEW

**Comment by atorok** [comment]
@elasticmachine test this please

**Comment by atorok** [comment]
@elasticmachine restest this please

File: modules/reindex/src/test/java/org/elasticsearch/index/reindex/remote/RemoteScrollableHitSourceTests.java [hunk 1]
```diff
@@ -150,15 +150,15 @@ public void testLookupRemoteVersion() throws Exception {
         assertTrue(called.get());
         called.set(false);
         sourceWithMockedRemoteCall(false, ContentType.APPLICATION_JSON, "main/5_0_0_alpha_3.json").lookupRemoteVersion(v -> {
-            // V_5_0_0_alpha3
-            assertEquals(Version.fromId(5000003), v);
+            // V_5_0_0 since we no longer consider qualifier in Version
```
Comment (nik9000) [inline modules/reindex/src/test/java/org/elasticsearch/index/reindex/remote/RemoteScrollableHitSourceTests.java]: Since 7 I think.

**Comment by atorok** [comment]
@elasticmachine test this please

**Comment by atorok** [comment]
@elasticmachine test this please

**Comment by atorok** [comment]
@nik9000 can you take another look ?

**Review by nik9000 — APPROVED** [review]
LGTM

**Pull request merged**